src/main/com/yetanalytics/datasim/input/model/alignments.clj

```diff
@@ -0,0 +1,147 @@
+(ns com.yetanalytics.datasim.input.model.alignments
+  (:require [clojure.set        :as cset]
+            [clojure.spec.alpha :as s]
+            [xapi-schema.spec   :as xs]
+            [com.yetanalytics.datasim.input.model.alignments.bound :as-alias bound]
+            [com.yetanalytics.datasim.input.model.alignments.delay :as-alias delay]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Weight
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; See also: ::object-override/weight
+
+(s/def ::weight
+  (s/double-in :min 0.0 :max 1.0 :infinite? false :NaN? false))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Time Bounds
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- interval? [[start end]]
+  (< start end))
+
+(defmacro bound-spec [scalar-spec]
+  `(s/every (s/or :scalar   ~scalar-spec
+                  :interval (s/and (s/tuple ~scalar-spec ~scalar-spec)
+                                   interval?))
+            :kind vector?
+            :min-count 1
+            :gen-max 3))
+
+(defmacro named-time-spec [index-spec name-index-map]
+  `(s/and
+    (s/or :integer ~index-spec
+          :string  (s/and (set (keys ~name-index-map))
+                          (s/conformer ~name-index-map
+                                       (cset/map-invert ~name-index-map))
+                          ~index-spec))
+    ;; Remove s/or tags
+    (s/conformer second)))
+
+(def day-of-week-map
+  {"Sunday"    0
+   "Monday"    1
```

Comment:
Does ISO 8601 start the week on Sunday or Monday? Might want to consider doing whatever it does

Reply:
ISO 8601 starts with...Monday...and is 1-indexed. Boo.

Reply:
The reason I chose 0 for Sunday is because that is what cron syntax does (and this time bound syntax is based off of cron)

Reply:
That's good enough for me. Good 'ol cron.